src/main/java/org/cloudfoundry/promregator/config/Target.java

```diff
@@ -203,6 +215,8 @@ public String toString() {
 		builder.append(applicationName);
 		builder.append(", applicationRegex=");
 		builder.append(applicationRegex);
+		builder.append(", applicationId=");
+		builder.append(applicationId);
```

Comment:
I added 'applicationId' to the `Target` and leveraged it to fix the failing test. I didn't add the new property to `equals` or `hashCode` since they hadn't been overridden in `Target` yet.

Reply:
I think adding `applicationId` is dangerous: It permits users to explicitly set the applicationId in their promregator configuration file (NB: `org.cloudfoundry.promregator.config.Target` is used as POJO for configuration serialization/deserialization!)

Reply:
It is dangerous. However it is the only was to have `TrivialTargetResolver` resolve the appId. I could fix the failing test by replacing `TrivialTargetResolver` with a mock that provides the correct IDs, but that indicated that `TrivialTargetResolver` doesn't work anymore and we probably should delete it.

Again, the other option of having the fallback to looking up the appId via the SpaceSummary would allow us to not add appId to Target, but that seemed like you wanted to avoid that.

Reply:
I haven't made a change here yet, since I'm not certain how you want me to proceed. I don't consider it a dangerous change for the users to be able to define the applicationId in their config, since it is ignored in the typical case and is now required if they are using the `TrivialTargetResolver` (I'm not certain how that resolver get activated, or if it is merely a testing helper)

If `TrivialTargetResolver` is only intended for testing, let me know. I will happily remove it and update the tests to no longer need it. That way I can remove the `applicationId` from the `Target` class.

Reply:
> If TrivialTargetResolver is only intended for testing, let me know.

Given that `TrivialTargetResolver` is under `src/test/java` I consider it as "test-only" coding. 
I would go even one step further: If the tests can no longer be performed with a `TrivialTargetResolver` (to whatever reason), then this class has reached its end-of-life and shall be replaced with "something more suitable". 

>  I don't consider it a dangerous change for the users to be able to define the applicationId in their config

I see it as false sense of contract: In fact, this attribute is intended to be "internal only" - we don't really mean anyone to tell us the app's id we shall scrape --- and even if they did, we still would have to read the same set of metadata as before, as we require the app's name as well (e.g. for Enrichment).
In short, I don't like that we extend a public contract with an attribute that we don't mean like this. I clearly prefer the way of putting `TrivialTargetResolver` through the mill instead.

Agreed?

Reply:
Wow. How did I overlook `TrivialTargetResolver` was a test only mechanism?  I'm happy to get rid of it and revert my changes to `Target`! I'll get right on that fix.

Reply:
thx a million!